### PR TITLE
[JENKINS-62297] Collect User counts

### PR DESCRIPTION
**Pull request by Dohbedoh**
[JENKINS-62297](https://issues.jenkins-ci.org/browse/JENKINS-62297) Add a component that provide User count to the support bundle.

### CODE REVIEW

File: src/main/java/com/cloudbees/jenkins/support/impl/UsersContent.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@Extension
+public class UsersContent extends UnfilteredFileListCapComponent {
+
+    @NonNull
+    @Override
+    public Set<Permission> getRequiredPermissions() {
+        return Collections.emptySet();
```
Comment (varyvol) [inline src/main/java/com/cloudbees/jenkins/support/impl/UsersContent.java:48]: This is my only doubt. Even if we are only displaying a number, is it something we want to do without checking permissions? 

@daniel-beck maybe you can provide your opinion.

File: src/main/java/com/cloudbees/jenkins/support/impl/UsersContent.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    }
+
+    @NonNull
+    @Override
+    public String getDisplayName() {
+        return "Users Count";
```
Comment (rsandell) [inline src/main/java/com/cloudbees/jenkins/support/impl/UsersContent.java:54]: A bit of strange English here. Should either be "User count" or "Number of users" if you want to use the plural form.

File: src/main/java/com/cloudbees/jenkins/support/impl/UsersContent.java [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+import jenkins.security.LastGrantedAuthoritiesProperty;
+
+import java.io.PrintWriter;
+import java.util.Collections;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@Extension
+public class UsersContent extends UnfilteredFileListCapComponent {
```
Comment (rsandell) [inline src/main/java/com/cloudbees/jenkins/support/impl/UsersContent.java:43]:
```suggestion
public class UserCount extends UnfilteredFileListCapComponent {
```

File: src/main/java/com/cloudbees/jenkins/support/impl/UsersContent.java (hunk 1 above)
Comment (Dohbedoh) [reply src/main/java/com/cloudbees/jenkins/support/impl/UsersContent.java]: Agreed. I think this needs Administer permissions. I'll let Daniel confirm.

File: src/main/java/com/cloudbees/jenkins/support/impl/UsersContent.java [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+
+                User.getAll().stream()
+                    .collect(
+                        Collectors.groupingBy(
+                            (user) -> Optional.ofNullable(user.getProperty(LastGrantedAuthoritiesProperty.class)).isPresent(),
+                            Collectors.counting()))
+                        .forEach((lastGrantedAuthoritiesProperty, aLong) ->
+                            out.println(" * "
+                                    + (lastGrantedAuthoritiesProperty ? "Authenticated" : "Non Authenticated")
+                                    + " Users count: " + aLong)
```
Comment (daniel-beck) [inline src/main/java/com/cloudbees/jenkins/support/impl/UsersContent.java:75]:
```suggestion
                                    + " Users Count: " + aLong)
```

The rest of this is title cased.

File: src/main/java/com/cloudbees/jenkins/support/impl/UsersContent.java [hunk 5]
```diff
@@ -0,0 +1,90 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2019, CloudBees, Inc.
```
Comment (daniel-beck) [inline src/main/java/com/cloudbees/jenkins/support/impl/UsersContent.java:4]:
```suggestion
 * Copyright (c) 2020, CloudBees, Inc.
```

File: src/main/java/com/cloudbees/jenkins/support/impl/UsersContent.java (hunk 3 above)
Comment (daniel-beck) [reply src/main/java/com/cloudbees/jenkins/support/impl/UsersContent.java:43]: It is a _Component_ adding a _Content_ to a bundle, so the current name seems misleading. (I would argue that `ItemsContent` has a misleading name too.)

File: src/main/java/com/cloudbees/jenkins/support/impl/UsersContent.java (hunk 4 above)
Comment (varyvol) [reply src/main/java/com/cloudbees/jenkins/support/impl/UsersContent.java:75]: It should actually "User Count" to match the component name.

Comment (daniel-beck) [reply src/main/java/com/cloudbees/jenkins/support/impl/UsersContent.java:75]: The point here is the uppercasing.

Comment (varyvol) [reply src/main/java/com/cloudbees/jenkins/support/impl/UsersContent.java:75]: Yes, I know, I was just adding something else to your proposal.

Comment (daniel-beck) [reply src/main/java/com/cloudbees/jenkins/support/impl/UsersContent.java:75]:
```suggestion
                                    + " User Count: " + aLong)
```

To make it official 😄

**Pull request merged**